src/phpDocumentor/ConfigurationFactory.php

```diff
@@ -0,0 +1,354 @@
+<?php
+
+namespace phpDocumentor;
+
+final class ConfigurationFactory
+{
+    /**
+     * @var Uri
+     */
+    private $uri;
+
+    /**
+     * @var string
+     */
+    private $schemaPath;
+
+    /**
+     * @param Uri    $uri
+     * @param string $schemaPath
+     */
+    public function __construct(Uri $uri, $schemaPath = '../../data/xsd/phpdoc.xsd')
```

Comment:
Please move that to the constructor; the $schemaPath may be an absolute path so you cannot prefix it with `__DIR__` by default.